packages/flutter/lib/src/material/dialog.dart

```diff
@@ -469,6 +469,8 @@ class AlertDialog extends StatelessWidget {
     String? label = semanticLabel;
     switch (theme.platform) {
       case TargetPlatform.iOS:
+        label = '';
```

Comment:
This isn't intuitive why it is the correct thing to do, we should probably add a small comment.